### PR TITLE
Fatpack service+fetch+blockers

### PR DESCRIPTION
**Pull request by atoomic**
Isolate all blockers to their namespace

### CODE REVIEW

**Comment by atoomic** [comment]
Still need some extra testing and need to consider after #228

File: lib/Elevate/Blockers.pm [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+sub _build_blockers { [] }
+
+sub check ($self) {    # do_check - main  entry point
+
+    if ( $self->cpev->service->is_active ) {
+        WARN("An elevation process is already in progress.");
+        return 1;
+    }
+
+    Elevate::Blockers::Distros::bail_out_on_inappropriate_distro();
```
Comment (toddr) [inline lib/Elevate/Blockers.pm:64]: why did we do this separately?

File: lib/Elevate/Blockers.pm [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            /scripts/elevate-cpanel --start
+        EOS
+    }
+
+    return $has_blockers;
+}
+
+sub _has_blockers ( $self, $check_mode = 0 ) {
+
+    unless ( $< == 0 ) {
```
Comment (toddr) [inline lib/Elevate/Blockers.pm:91]: should be in a namespace.

File: lib/Elevate/Blockers/Distros.pm [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    unless ( Cpanel::OS::major() == 7 && Cpanel::OS::distro() eq 'centos' ) {
+        my $pretty_distro_name = $self->upgrade_to_pretty_name();
+        return $self->has_blocker(qq[This script is only designed to upgrade CentOS 7 to $pretty_distro_name.]);
+    }
+
+    return 0;
+}
+
+sub _blocker_is_old_centos7 ($self) {
+    if ( Cpanel::OS::minor() < Elevate::Constants::MINIMUM_CENTOS_7_SUPPORTED ) {
```
Comment (toddr) [inline lib/Elevate/Blockers/Distros.pm:39]: constants should be local?

**Pull request merged**